service/internal/proctelemetry/config.go

```diff
@@ -0,0 +1,244 @@
+	"go.opentelemetry.io/collector/service/telemetry"
+)
+
+const (
+	// supported exporters
+	consoleExporter      = "console"
+	otlpExporter         = "otlp"
+	protocolProtobufHTTP = "http/protobuf"
+	protocolProtobufGRPC = "grpc/protobuf"
+	compressionGzip      = "gzip"
+
+	// supported metric readers
+	PrometheusMetricReader = "prometheus"
+	PeriodMetricReader     = "periodic"
+
+	// gRPC Instrumentation Name
+	GRPCInstrumentation = "go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+
+	// http Instrumentation Name
+	HTTPInstrumentation = "go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+)
+
+var (
+	// GRPCUnacceptableKeyValues is a list of high cardinality grpc attributes that should be filtered out.
+	GRPCUnacceptableKeyValues = []attribute.KeyValue{
+		attribute.String(semconv.AttributeNetSockPeerAddr, ""),
+		attribute.String(semconv.AttributeNetSockPeerPort, ""),
+		attribute.String(semconv.AttributeNetSockPeerName, ""),
+	}
+
+	// HTTPUnacceptableKeyValues is a list of high cardinality http attributes that should be filtered out.
+	HTTPUnacceptableKeyValues = []attribute.KeyValue{
+		attribute.String(semconv.AttributeNetHostName, ""),
+		attribute.String(semconv.AttributeNetHostPort, ""),
+	}
+)
+
+func toStringMap(in map[string]interface{}) map[string]string {
+	out := map[string]string{}
+	for k, v := range in {
+		out[k] = fmt.Sprintf("%v", v)
+	}
+	return out
+}
+
+func newOTLPGRPCMetricExporter(ctx context.Context, args map[string]interface{}) (sdkmetric.Exporter, error) {
+	opts := []otlpmetricgrpc.Option{}
+	for k, v := range args {
+		switch k {
+		case "endpoint":
+			opts = append(opts, otlpmetricgrpc.WithEndpoint(fmt.Sprintf("%s", v)))
+		case "certificate":
+		case "client_key":
+		case "client_certificate":
+		case "compression":
+			opts = append(opts, otlpmetricgrpc.WithCompressor(fmt.Sprintf("%s", v)))
+		case "timeout":
+			timeout, ok := v.(int)
+			if !ok {
+				return nil, fmt.Errorf("invalid timeout for otlp exporter: %s", v)
+			}
+			opts = append(opts, otlpmetricgrpc.WithTimeout(time.Millisecond*time.Duration(timeout)))
+		case "headers":
+			headers, ok := v.(map[string]any)
+			if !ok {
+				return nil, fmt.Errorf("invalid headers for otlp exporter: %s", v)
+			}
+			opts = append(opts, otlpmetricgrpc.WithHeaders(toStringMap(headers)))
+			// otlpmetricgrpc.WithInsecure()
+			// otlpmetricgrpc.WithReconnectionPeriod()
+			// otlpmetricgrpc.WithTLSCredentials()
+			//
+		}
+	}
+	return otlpmetricgrpc.New(ctx, opts...)
+}
+
+func newOTLPHTTPMetricExporter(ctx context.Context, args map[string]interface{}) (sdkmetric.Exporter, error) {
+	opts := []otlpmetrichttp.Option{}
+	for k, v := range args {
+		switch k {
+		case "endpoint":
+			opts = append(opts, otlpmetrichttp.WithEndpoint(fmt.Sprintf("%s", v)))
+		case "certificate":
+		case "client_key":
+		case "client_certificate":
+		case "compression":
+			switch fmt.Sprintf("%s", v) {
```

Comment:
At this point you can't assert that `v` is a string due to args being a map of string to any, switching to `fmt.Sprint` should work without issue here.